### DISCUSSION TITLE
FourThree Utility: Installation appears broken (multiple issues, read for details)

### DISCUSSION THREAD

**Discussion by JosephM101** [LukeZGD/Legacy-iOS-Kit#509] [General]
I'm trying to get FourThree working on my iPad2,3. I've tried the same process five times, but it has failed every time. I've tried with both iOS 4.3.3 and 4.3.5 as the base firmware (did I choose the wrong ones? Please tell me...) I have been running this script on Linux Mint.

Here are the issues I have:

- In the iOS 6.1.3 environment, after FourThree is installed, _all_ apps (except for the FourThree launcher and Newsstand) disappear from the home screen, rendering the iOS 6.1.3 environment unusable.
- When booting into the iOS 4.3.x environment, the device boots to a screen asking to be plugged into iTunes for first setup. However, both Finder on the Mac and iTunes (on macOS and Windows) remain stuck on a "Loading" view/spinner when trying to identify the iPad.
  - The first time I attempted this, Finder on the Mac did briefly manage to identify the iPad, but showed an error saying it was unable to activate the device.
- I cannot activate the iPad using the Legacy iOS Kit. I get an error saying that the activation server failed.
- After about 2-3 minutes of being booted into the iOS 4.3.x environment, the iPad seems to crash and reboot into iOS 6.1.3. Based on the other issues above, I'm assuming that this could be related to some sort of mount-point/partitioning failure.

Here is what I've been doing:
- In Legacy iOS Kit, with the iPad plugged in to USB and jailbroken with OpenSSH and coreutils installed, I begin Step 1 of the FourThree Utility, and restore the iPad to iOS 6.1.3.
   - **NOTE**: Before each _full_ attempt at installing FourThree, I make sure to delete the 'saved' and 'tmp' directories to make sure all leftovers from previous attempts are cleaned.
- In Legacy iOS Kit, I go to `Main Menu > FourThree Utility > Step 1: Restore`. I then choose option `2) Download Target IPSW` to download the iOS 6.1.3 IPSW from Apple. I then go to ipsw.me to download the iOS 4.3.x IPSW for my iPad2,3, and in Legacy iOS Kit, I choose `3) Select Base IPSW` and select the iOS 4.3.x IPSW that I just downloaded.
- After completing Option 3, I am brought back to `Main Menu > FourThree Utility > Step 1: Restore`. Option `4) Start Restore` is now available. I choose Option 4, and follow the instructions, putting the iPad into `kDFU` mode and allowing the restore to complete.
  - I do not change the default values of any of the config prompts that appear.
  - I have had a few issues before when entering kDFU mode, such as the iPad simply not being detected. I personally don't think that 5 attempts is enough, and would recommend a change that allows a user to specify how many connection retries should be attempted, maximum, when running Legacy iOS Kit. (ex. `--kdfu-connect-max-retries $NUM_OF_RETRIES`)
- After the restore completes, I set up the iPad as usual and connect it to WiFi. I do not sign in with an Apple ID. 
- Once the iOS onboarding setup is complete, I open Cydia, and allow it to "prepare the filesystem".
- After Cydia initializes and re-springs the iPad, I then allow Cydia to update its sources, and I make sure the following packages are installed (as per what Legacy iOS Kit requests before entering kDFU mode):
  - `OpenSSH` (usually already installed)
  - `Core Utilities`
- I relaunch the Legacy iOS Kit, and choose `Main Menu > FourThree Utility > Step 2: Partition`.
  - For `[Input] iOS 6.1.3 Data Partition Size (in GB)`, I specify 8 GB.
- After the `Partition` step completes and the iPad reboots, I relaunch Legacy iOS Kit, and choose `Main Menu > FourThree Utility > Step 3: OS Install`. The script appears to initialize the target partitions and copy the iOS 4 root filesystem.
  - **NOTE**: Just before this step completes is where I observe all apps (except for Newsstand and FourThree) disappear from the home screen: When the script runs the `uicache` command. In one or two cases, the Cydia app remained, but its icon disappeared and any attempts to open it failed. It's almost like the iOS 6 root partition isn't mounted (properly) anymore; I've seen similar behavior on Linux systems.

Has anybody else attempted to install FourThree onto their iPad 2? Have you done so with success?
I am currently in the process of redoing the entire setup from scratch and saving the logs from each step. I will attach them here once finished, and sensitive info is removed.

I would wholeheartedly appreciate any guidance. Sorry this post is so long, but I wanted to be as detailed as possible. Thank you for reading.

**Comment by JosephM101** [comment]
Here are my log files from Legacy iOS Kit:

[step1_logs.txt](https://github.com/user-attachments/files/15855894/step1_logs.txt)
[step2_logs.txt](https://github.com/user-attachments/files/15855895/step2_logs.txt)
[step3_logs.txt](https://github.com/user-attachments/files/15855896/step3_logs.txt)

Thank you

**Comment by LukeZGD** [comment]
> In the iOS 6.1.3 environment, after FourThree is installed, all apps (except for the FourThree launcher and Newsstand) disappear from the home screen, rendering the iOS 6.1.3 environment unusable.

I haven't seen this issue when testing, though I actually have not tested opening Cydia on 6.1.3 since OpenSSH is pre-installed on Step 1, idk if this affects things tho

> I cannot activate the iPad using the Legacy iOS Kit. I get an error saying that the activation server failed.

Yeah there are activation issues on the cellular devices, this is also said to be an issue in 4tify-iPad2

There might be a way to get past this by just copying over the 6.1.3 activation records, I'll try on a later date on my 2,1

> After about 2-3 minutes of being booted into the iOS 4.3.x environment, the iPad seems to crash and reboot into iOS 6.1.3. Based on the other issues above, I'm assuming that this could be related to some sort of mount-point/partitioning failure.

idk, I also haven't encountered this issue myself

In any case though, FourThree-iPad2 is only tested on iPad2,1 where most of these issues are likely to be absent

From this issue, it can be said that FourThree currently has bad compatibility with the cellular devices, idk if this will ever be fixed since I don't have test devices for it, nor much interest in updating the FourThree utility part of the kit tbh.. For now, the better (yet a lot more inconvenient) option is currently to tether downgrade to 4.3.x

**Comment by JosephM101** [comment]
@LukeZGD Thank you for your timely response. It's a bummer that this doesn't work on the cellular versions, but I understand it's not your fault, and I respect your decision to not update the FourThree utility.

> For now, the better (yet a lot more inconvenient) option is currently to tether downgrade to 4.3.x

If I'm being honest, I don't intend to stick with iOS 4 on the iPad since it really can't do much for me long-term. But I'm really obsessed with the idea of seeing the spectrum analyzer in the music app, like the one that got removed from iTunes? I (think I) saw it once on a school teacher's iPad during gym class back in elementary school and thought it was the coolest thing ever. So if I have to connect to a PC or something to boot, then so be it. I plan to do iOS 8 with iOS 6 dual-boot using Coolbooter in the future. I just want to experience iOS 4 for a bit.

Would you mind explaining to me how to do the tethered boot?

**Comment by LukeZGD** [comment]
Tether downgrading can be done by going to Restore/Downgrade and selecting the "Other (Tethered)" option: https://github.com/LukeZGD/Legacy-iOS-Kit/wiki/Restore-Downgrade

Here's the catch, though - since you have an A5 device, you will need an Arduino+USB Host Shield with checkm8-a5 to boot it (Other Utilities -> Just Boot): https://github.com/LukeZGD/Legacy-iOS-Kit/wiki/Troubleshooting#dfu-advanced-menu-a5x-pwndfu-mode-with-arduino-and-usb-host-shield

This is why I said that it is a lot more inconvenient

**Comment by JosephM101** [comment]
I didn't know I needed hardware... That sucks.

I have a Raspberry Pi Pico, which has built-in USB Host support. If I hook it up to a battery and use a USB-OTG adapter, it could *theoretically* work.
Does this exploit only exist for Arduino?

**Comment by LukeZGD** [comment]
I think there is a checkm8-a5 for the pico, idk if it will work since I have seen reports that it does not work properly for downgrade/boot purposes

**Reply by JosephM101** [reply]
I guess this is a dead end, then. I have an Arduino, but am unable to buy an OEM USB Host Shield, nor do I deem it a "wise purchase" for my intended purposes.

Anyways, thank you very much for clearing this all up for me. I really appreciate it. Inconveniences aside, this is a really awesome project!

**Comment by LukeZGD** [comment]
checkm8-a5 for the pico is now an option btw if youre interested: https://www.reddit.com/r/LegacyJailbreak/comments/1djuprf/working_checkm8a5_on_the_raspberry_pi_pico/

**Comment by LukeZGD** [comment]
activation issues with fourthree utility are now fixed

as for any other issues, idk what the status is since i havent encountered any of them

**Reply by JosephM101** [reply]
Thank you, but I installed iOS 4 using checkm8-a5 and had the same issues. No activation, and the iPad would just crash and reboot. My iPad is a CDMA model, and might not have shipped with iOS 4, either. I'm not sure, nor am I sure how to find out.

Should I try anyway?

**Reply by LukeZGD** [reply]
while the activation issue is solved, i dont think it is worth it because of the crash issue making it unusable

maybe it is something with the ipad 2 cdma, and/or the patches used, im not sure
